src/didemo_eval.py

```diff
@@ -0,0 +1,66 @@
+from aggregation.mean import Mean
+import sys
+import os
+import open_clip
+import torch
+
+sys.path.insert(1, '/Users/daniel/Desktop/LAION_Videoclip/clip-video-encode')
+from clip_video_encode.dataset import EmbeddingWebDatasetReader
+
+from evaluation.retrieval import retrieval_evaluation
+
+def process_times(times, len_embeddings):
+    '''
+    Assumptions: 
+    - times contains [start, end] in intervals of 5 seconds, 
+    i.e. [0, 1] corresponds to [0*5, (1+1)*5]
+
+    - there's 1 embedding per second, i.e. embeddings[10] is the sole embedding for the 10th second
+    '''
+    SEGMENT_INTERVAL = 5
+    return [
+        SEGMENT_INTERVAL * times[0], 
+        min( SEGMENT_INTERVAL * (times[1] + 1), len_embeddings )
+    ]
+
+def zero_pad(e, seq_len, model_dim=1024):
+    out = torch.zeros(size=(seq_len, model_dim))
+    out[0:len(e)] = e
+
+    return out
+
+def process_didemo_segments(embeddings, segments, seq_len=200):
```

Comment:
no function, no specific code for each dataset, just move these operations into the "if "segments" in batch["meta"]" part of the retrieval eval